src/main/java/org/fenixedu/academic/domain/student/RegistrationDataByExecutionYear.java

```diff
@@ -28,21 +32,54 @@
  */
 public class RegistrationDataByExecutionYear extends RegistrationDataByExecutionYear_Base {
 
+    // This method must be deleted (or at least made private) to avoid misusage 
```

Comment:
As this PR is targeted to a major version, it's OK to delete those methods (or make them private).